pilot/helpers/agents/Developer.py

```diff
@@ -92,38 +91,42 @@ def implement_task(self, i, development_task=None):
             'running_processes': running_processes,
             'os': platform.system(),
         }, IMPLEMENT_TASK)
-        task_steps = response['tasks']
+        steps = response['tasks']
         convo_dev_task.remove_last_x_messages(2)
 
+        completed_steps = []
+
         while True:
             result = self.execute_task(convo_dev_task,
-                                     task_steps,
+                                     steps,
                                      development_task=development_task,
                                      continue_development=True,
-                                     is_root_task=True)
+                                     is_root_task=True,
+                                     continue_from_step=len(completed_steps))
 
             if result['success']:
                 break
 
             if 'step_index' in result:
-                result['running_processes'] = running_processes
                 result['os'] = platform.system()
                 step_index = result['step_index']
-                result['completed_steps'] = task_steps[:step_index]
-                result['current_step'] = task_steps[step_index]
-                result['next_steps'] = task_steps[step_index + 1:]
-                result['current_step_index'] = step_index
+                completed_steps = steps[:step_index+1]
+                result['completed_steps'] = completed_steps
+                result['current_step'] = steps[step_index]
+                result['next_steps'] = steps[step_index + 1:]
+                result['current_step_index'] = step_index + len(completed_steps)
```

Comment:
Is this correct? Your change to `execute_task` (*Skip steps before continue_from_step*) means the step index there (`i`) will start from 0, skip over the completed steps (because of `continue_from_step`), and the counter will already include the count of all completed steps.

Reply:
nice catch!